zipline/pipeline/loaders/blaze/core.py

```diff
@@ -312,21 +311,17 @@ def new_dataset(expr, deltas, missing_values):
         if name in (SID_FIELD_NAME, TS_FIELD_NAME):
             continue
         try:
-            # TODO: This should support datetime and bool columns.
-            if promote(type_, float64, promote_option=False) != float64:
-                raise NotPipelineCompatible()
             if isinstance(type_, Option):
                 type_ = type_.ty
-        except NotPipelineCompatible:
-            col = NonPipelineField(name, type_)
-        except TypeError:
-            col = NonNumpyField(name, type_)
-        else:
+            type_ = type_.to_numpy_dtype()
```

Comment:
If we're not catching `TypeError` anymore, then the weird goto here isn't necessary anymore.  This is just:

```
type_ = type_.to_numpy_dtype()
if can_represent_dtype(type_):
    col = Column(...)
else:
    col = NonPipelineField(...)
```